File: v1/src/main/java/com/api/v1/dtos/CustomerUpdatingRequestDto.java
package com.api.v1.dtos;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

public record CustomerUpdatingRequestDto(
        @NotBlank String firstName,
        String middleName,
        @NotBlank String lastName,
        @NotNull LocalDate birthDate,
        @NotBlank @Email String email,
        @NotBlank String address,
        @NotBlank @Size(min=10, max=10) String phoneNumber
) {
}

File: v2/src/main/kotlin/com/api/v2/CustomerUpdatingRequestDto.kt
package com.api.v2

import java.time.LocalDate
import java.util.*

data class CustomerUpdatingRequestDto(
    val phoneNumber: String,
    val address: String,
    val email: String,
    val birthDate: LocalDate,
    val lastName: String,
    val middleName: String,
    val firstName: String
)
